src/main/java/org/openlmis/buq/web/productgroup/ProductGroupController.java

```diff
@@ -0,0 +1,178 @@
+import org.openlmis.buq.dto.productgroup.ProductGroupDto;
+import org.openlmis.buq.exception.NotFoundException;
+import org.openlmis.buq.exception.ValidationMessageException;
+import org.openlmis.buq.i18n.MessageKeys;
+import org.openlmis.buq.repository.productgroup.ProductGroupRepository;
+import org.openlmis.buq.util.Pagination;
+import org.openlmis.buq.web.BaseController;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+/**
+ * Controller used to expose Product groups via HTTP.
+ */
+@Controller
+@RequestMapping(ProductGroupController.RESOURCE_PATH)
+@Transactional
+public class ProductGroupController extends BaseController {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(ProductGroupController.class);
+
+  public static final String RESOURCE_PATH = API_PATH + "/productGroups";
+
+  @Autowired
+  private ProductGroupRepository productGroupRepository;
+
+  /**
+   * Allows the creation of a new product group. If the id is specified, it will be ignored.
+   */
+  @PostMapping
+  @ResponseStatus(HttpStatus.CREATED)
+  @ResponseBody
+  public ProductGroupDto createProductGroup(@RequestBody ProductGroupDto productGroup) {
+    LOGGER.debug("Creating new product group");
+    ProductGroup newProductGroup = ProductGroup.newInstance(productGroup);
+    newProductGroup.setId(null);
+    newProductGroup = productGroupRepository.saveAndFlush(newProductGroup);
+
+    return ProductGroupDto.newInstance(newProductGroup);
+  }
+
+  /**
+   * Updates the specified product group.
+   */
+  @PutMapping(value = "/{id}")
+  @ResponseStatus(HttpStatus.OK)
+  @ResponseBody
+  public ProductGroupDto saveProductGroup(@PathVariable("id") UUID id,
+                                          @RequestBody ProductGroupDto productGroup) {
+    if (null != productGroup.getId() && !Objects.equals(productGroup.getId(), id)) {
+      throw new ValidationMessageException(MessageKeys.ERROR_PRODUCT_GROUP_ID_MISMATCH);
+    }
+
+    LOGGER.debug("Updating product group");
```

Comment:
Please add productGroup.code to the log.

Reply:
Code added.